core/test/gather/gatherers/full-page-screenshot-test.js

```diff
@@ -201,4 +207,338 @@ describe('FullPageScreenshot gatherer', () => {
       }
     );
   });
+
+  // Tests that our node rects line up with content in the screenshot image data.
+  // This uses "screenshot-nodes.html", which has elements of solid colors to make verification simple.
+  // To verify a node rect is correct, each pixel in its area is looked at in the screenshot data, and is checked
+  // for the expected color. Due to compression artifacts, there are thresholds involved instead of exact matches.
+  describe('end-to-end integration test', () => {
+    const port = 10503;
+    let serverBaseUrl = ''
+    /** @type {StaticServer} */;
+    let server;
+    /** @type {puppeteer.Browser} */
+    let browser;
+    /** @type {puppeteer.Page} */
+    let page;
+
+    before(async () => {
+      browser = await puppeteer.launch({
+        headless: true,
+        executablePath: getChromePath(),
+        ignoreDefaultArgs: ['--enable-automation'],
+      });
+
+      server = new Server(port);
+      await server.listen(port, '127.0.0.1');
+      serverBaseUrl = `http://localhost:${server.getPort()}`;
+    });
+
+    after(async () => {
+      await browser.close();
+      await server.close();
+    });
+
+    beforeEach(async () => {
+      page = await browser.newPage();
+    });
+
+    afterEach(async () => {
+      await page.close();
+    });
+
+    /**
+     * @typedef NodeAnalysisResult
+     * @property {string} id
+     * @property {boolean} success
+     * @property {number[][]} debugData
+     */
+
+    /**
+     * @param {puppeteer.Page} page
+     * @param {LH.Result} lhr
+     * @param {DebugFormat} debugFormat
+     * @return {Promise<NodeAnalysisResult>}
+     */
+    function analyzeScreenshotNodes(page, lhr, debugFormat) {
+      const options = {
+        fullPageScreenshot: lhr.fullPageScreenshot,
+        // We currently can't get lossless encodings here, so we must allow for some difference.
+        // I found 2.4 is what I get with the test data for regions that match, and 100+ for regions that are clearly not matching.
+        // So using 5 seems a good bet.
+        // If we could get lossless webp/png here, this should go away.
+        // TODO https://bugs.chromium.org/p/chromium/issues/detail?id=1469183
```

Comment:
https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-captureScreenshot

png should be supported, is that lossy as well?